hooks/hostname.hook/README.md

```diff
@@ -1,49 +1,45 @@
 ## Sample counting hook
 
+You can use a hook to create more advanced dynamic hostnames than the simple incremented pattern --- `$\{id\}.example.com` --- from the `hostname` property on a policy. This type of hook calculates the correct hostname and returns that hostname as metadata on the task. To do so, it uses a basic counter system that stores how many nodes have bound to a given policy.
+
 ### Prerequisites
 
-This hook requires that ruby is installed. This must be contained in $PATH in
-order for the hook script to succeed.
+Ruby must be installed in `$PATH` for the hook script to succeed.
 
-### Purpose
+### Install the Hook
 
-This hook performs dynamic hostname assignment. It uses a basic counter system,
-storing how many nodes have bound to a given policy.
+This hook comes with Razor. To use it, create an instance of
+the hook with the following command:
 
-### How to install
 
-This hook comes with Razor; all you will need to do is create an instance of
-the hook via:
+        razor create-hook --name some_policy_hook --hook-type hostname \
+            --configuration policy=some_policy \
+            --configuration hostname-pattern='${policy}${count}.example.com'
 
-```
-razor create-hook --name some_policy_hook --hook-type hostname \
-                  --configuration policy=some_policy \
-                  --configuration hostname-pattern='${policy}${count}.example.com'
-```
 
-### How it works
+### How It Works
 
-The order of events is as follows:
+Running the above `create-hook` command kicks off the following sequence of events:
 
 1. The counter for the policy starts at 1.
 2. When a node boots, the `node-bound-to-policy` event is triggered.
 3. The policy's name from the event is then passed to the hook as input.
 4. The hook matches the node's policy name to the hook's policy name.
 5. If the policy matches, the hook calculates a rendered hostname pattern:
-   - It replaces `${count}` with the current value of the `counter` hook 
+   - It replaces `${count}` with the current value of the `counter` hook
      configuration.
-   - It left-pads the `${count}` with `padding` zeroes. E.g. If the hook
+   - It left-pads the `${count}` with `padding` zeroes. For example, if the hook
      configuration's `padding` equals 3, a `count` of 7 will be rendered as
      `007`.
    - It replaces `${policy}` with the name of its policy.
-6. The hook then returns the rendered hostname-pattern as the node metadata
+6. The hook then returns the rendered hostname pattern as the node metadata
    of `hostname`.
 7. The hook also returns the incremented value for the counter that was used.
```

Comment:
"The hook also returns the incremented value for the counter that was used so that the next execution of the hook will use the next value."